packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -129,6 +131,21 @@ export class ExtensionService {
                 );
             }
         });
+
+        Object.values(queryHandlerFactories).forEach((createQueryHandlerFactory) => {
+            const handlerFactory = createQueryHandlerFactory(handlerProps);
+
+            if (this.isQueryHandler(handlerFactory.queryType, handlerFactory)) {
+                this.handlers[extension.id].push(
+                    this.checkoutService.handleExtensionQuery(
+                        extension.id,
+                        handlerFactory.queryType,
+                        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+                        handlerFactory.handler,
+                    ),
```

Comment:
We currently define query handler the same way as the command handler.
```
handler: (command: ExtensionQueryMap[T]) => void,
```
However, due to the nature of making API calls for a query, we may consider change it to below.
```
handler: (command: ExtensionQueryMap[T]) => Promise<void>,
```
```